jdaviz/configs/imviz/helper.py

```diff
@@ -354,14 +390,42 @@ def split_filename_with_fits_ext(filename):
     return filepath, ext, data_label
 
 
-def layer_is_image_data(layer):
+def layer_is_2d(layer):
+    # returns True for subclasses of BaseData with ndim=2, both for
+    # layers that are WCS-only as well as images containing data:
     return isinstance(layer, BaseData) and layer.ndim == 2
 
 
+# NOTE: Sync with app._wcs_only_label as needed.
+def layer_is_image_data(layer):
+    return layer_is_2d(layer) and not layer.meta.get("_WCS_ONLY", False)
+
+
+# NOTE: Sync with app._wcs_only_label as needed.
+def layer_is_wcs_only(layer):
+    return layer_is_2d(layer) and layer.meta.get("_WCS_ONLY", False)
+
+
 def layer_is_table_data(layer):
     return isinstance(layer, BaseData) and layer.ndim == 1
 
 
+def get_bottom_layer(viewer):
```

Comment:
If we want to keep this functionality, I have decided to rename it ("bottom layer" means nothing to me) and not have to loop through the entire collection. And also move it out of helper to viewer because this is can be viewer property.

Reply:
And I don't think visibility matters here? Otherwise why not just use `get_top_layer_index`?  Removing that check as well.

Reply:
> get_top_layer_index gets the last-loaded visible image layer.

Not necessarily true after blinking. The top visible layer is whatever someone is seeing and it does not have to be the last loaded image.